cpg_workflows/inputs.py

```diff
@@ -35,187 +35,119 @@ def create_cohort() -> Cohort:
         )
-        for ds in cohort.get_datasets():
-            ds_sids_wo_seq = [sid for sid in sids_wo_seq if sid in ds.get_sample_ids()]
-            logging.info(
-                f'\t{ds.name}, {len(ds_sids_wo_seq)}/{len(ds.get_samples())} samples: '
-                f'{", ".join(ds_sids_wo_seq)}'
-            )
 
-    sid_wo_reads = set()
-    for sample in cohort.get_samples():
-        for entry in seq_entries_by_sid.get(sample.id, []):
-            seq = Sequence.parse(entry, parse_reads=False)
-            if seq.sequencing_type in sample.seq_by_type:
-                raise MetamistError(
-                    f'{sample}: found more than one associated sequencing entry with '
-                    f'sequencing type: {seq.sequencing_type}. Make sure there is only '
-                    f'one data source of sequencing type per sample.'
-                )
-            sample.seq_by_type[seq.sequencing_type] = seq
+    assay_entry = entry['assays'][0]
 
-            if not entry.get('meta', {}).get('reads'):
-                sid_wo_reads.add(sample.id)
-                continue
+    assay = Assay.parse(assay_entry, sequencing_group.id, parse_reads=False)
 
-            alignment_input = Sequence.parse_reads(
-                sample_id=sample.id,
-                meta=entry['meta'],
-                check_existence=check_existence,
-            )
-            seq.alignment_input = alignment_input
-            sample.alignment_input_by_seq_type[seq.sequencing_type] = alignment_input
+    sequencing_group.assays[assay.assay_type] = assay
 
-    if sid_wo_reads:
+    if assay_entry.get('meta', {}).get('reads'):
+        alignment_input = Assay.parse_reads(
+            sequencing_group_id=sequencing_group.id,
+            meta=assay_entry['meta'],
+            check_existence=check_existence,
+        )
+        assay.alignment_input = alignment_input
+        sequencing_group.alignment_input_by_seq_type[entry['type']] = alignment_input
+    else:
         logging.warning(
-            f'Found {len(sid_wo_reads)}/{len(cohort.get_samples())} samples with '
-            f'no meta/reads in corresponding sequence entries'
+            f'No reads found for sequencing group {sequencing_group.id} of type {entry["type"]}'
         )
-        for ds in cohort.get_datasets():
-            ds_sid_wo_reads = [
-                sid for sid in sid_wo_reads if sid in ds.get_sample_ids()
-            ]
-            logging.warning(
-                f'\t{ds.name}, {len(ds_sid_wo_reads)}/{len(ds.get_samples())} samples: '
-                f'{", ".join(ds_sid_wo_reads)}'
-            )
+
+    return None
 
 
 def _populate_analysis(cohort: Cohort) -> None:
     """
     Populate Analysis entries.
     """
     for dataset in cohort.get_datasets():
-        gvcf_by_sid = get_metamist().get_analyses_by_sid(
-            dataset.get_sample_ids(),
+        gvcf_by_sgid = get_metamist().get_analyses_by_sid(
```

Comment:
Were these functions going to change to `get_…_by_sgid()`? (I guess the other two have evaporated, so only `get_analyses_by_s[g]id()` remains.)